src/nvim/api/vim.c

```diff
@@ -119,12 +83,26 @@ Integer nvim_get_hl_id_by_name(String name)
   return syn_check_group(name.data, name.size);
 }
 
-Dictionary nvim__get_hl_defs(Integer ns_id, Arena *arena, Error *err)
+/// Gets highlight groups
+///
+/// @note Unlike the `:highlight` command which lists all the current highlight
+///       groups that have attributes set in human readable format, this function
+///       returns a table of all current highlight groups in a machine readable format
+///       (i.e. can be directly fed to nvim_set_hl)
+///
+/// @param ns_id Get highlight groups using namespace ns_id |nvim_get_namespaces()|.
+///              Use 0 to get global highlight groups |:highlight|.
+/// @param opts  Options dict:
+///                 - name: (string) Get a highlight definition by name.
+///                 - id: (integer) Get a highlight definition by id.
+///                 - link: (boolean, default true) Show highlight's link instead of definition.
+///
+/// @param[out] err Error details, if any.
+/// @return Highlight definition map (if `name` or `id` were provided), or map of all highlight groups in the given namespace.
```

Comment:
```suggestion
/// Gets all or specific highlight groups in a namespace.
///
/// @param ns_id Get highlight groups for namespace ns_id |nvim_get_namespaces()|.
///              Use 0 to get global highlight groups |:highlight|.
/// @param opts  Options dict:
///                 - name: (string) Get a highlight definition by name.
///                 - id: (integer) Get a highlight definition by id.
///                 - link: (boolean, default true) Show linked group name instead of effective definition |:hi-link|. 
///
/// @param[out] err Error details, if any.
/// @return Highlight groups as a map from group name to a highlight definition map as in |nvim_set_hl()|,
///                   or only a single highlight definition map if requested by name or id.
```